app/controllers/access_permissions/projects_controller.rb

```diff
@@ -161,6 +162,19 @@ def set_project
       render_404 unless @project
     end
 
+    def remove_access_from_all_team_members
+      users = @project.assigned_users.reject { |user| user == current_user }
+      users.each do |user|
+        user_assignment = @project.user_assignments.find_by(user: user, team: current_team)
+
+        if user_assignment.last_with_permission?(ProjectPermissions::USERS_MANAGE, assigned: :manually)
+          raise ActiveRecord::RecordInvalid
+        end
+
+        propagate_job(user_assignment, destroy: true)
+      end
+    end
```

Comment:
This logic should go into the Project model, so this happens any time the project visibility is updated.